src/main/java/com/google/graphgeckos/dashboard/storage/ParsedGitData.java

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.graphgeckos.dashboard.storage;
+
+/**
+ * An immutable container used for providing stripped-down git information
+ * to the StorageController. Provides functionality for checking whether the information
+ * is sufficient for creating an entry. Does not modify or validity check the individual
+ * contents of each field.
+ *
+ * Fields included: 
+ * -commitHash: the commit hash of the revision built by a certain build bot.
+ * -timestamp: the formatted date when the commit was pushed.
+ * -branch: the branch of the LLVM project on which this commit was pushed.
+ */
+public class ParsedGitData {
```

Comment:
LGTM
I'd suggest a name change for the class - "GitData" rather than ParsedGitData, to better communicate the fact that this represents the "Git Data" Entity.